tests/gpu/CMakeLists.txt

```diff
@@ -73,7 +73,14 @@ function(CREATE_GPU_TEST_RDC exe source)
         message(FATAL_ERROR "ENABLE_HIP or ENABLE_CUDA must be defined for GPU tests")
     endif()
 
-    add_test(NAME ${exe}.${lang}.rdc COMMAND ${LIT} -vv -D EXT=${lang}.rdc ${source})
+    # Code to pull only lit instructions from first source file
+    list(LENGTH source num_source_files)
+    if(${num_source_files} GREATER 1)
+      list(GET source 0 lit_instructions)
+      add_test(NAME ${exe}.${lang}.rdc COMMAND ${LIT} -vv -D EXT=${lang}.rdc ${lit_instructions})
+    else()
+      add_test(NAME ${exe}.${lang}.rdc COMMAND ${LIT} -vv -D EXT=${lang}.rdc ${source})
+    endif()
```

Comment:
Do we need this? We already have a multi-file test (`file1_kernel.cpp`, `file2_kernel.cpp`) and lit picks up checks from the first file

Reply:
On the rdc tests you do not have multi-file tests and are not taking into account the ARGN function parameter.

Now that both functions have an extra argument (will_fail) I bundle all sources together as a list on the second argument.